### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by maaloum**
In this project, I :

Linked front end and back end
Hit the backend API through the front end

### CODE REVIEW

**Review by descholar-ceo — CHANGES_REQUESTED** [review]
Hi @maaloum 🤝 , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that has a frontend repo that misses an open PR that adds changes of connecting with the backend, would you open a PR in the frontend repo so that we will be able to review your changes?

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: app/controllers/api/v1/greetings_controller.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
+class Api::V1::GreetingsController < ApplicationController
+  before_action :set_greeting, only: %i[show update destroy]
+
+  # GET /greetings
+  def index
+    @greetings = Greeting.all
+
+    render json: @greetings
+  end
```
Comment (AmaduKamara) [inline app/controllers/api/v1/greetings_controller.rb]: - [ ] Good job so far, but there is a little more work to be done. You are expected to return random greetings when this route is hit. At the moment, you're returning all the greetings and so does your frontend. Kindly work on returning random greeting(One greeting) whenever the page is refreshed 👍🏽

**Review by AmaduKamara — CHANGES_REQUESTED** [review]
# Status: Required Changes ♻️ ♻️ ♻️

Hi @maaloum,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✅ You used Gitflow correctly 👍🏽
✅ The Linter checks passed successfully 👍🏽

## Required Changes ♻️
_Check the comments under the review._

- [ ] Kindly work on returning random greetings and not all the greetings at once. 👍🏽
- [ ] Please refer to the frontend repo for required changes asked of you 👍🏽

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by GzimAsani — CHANGES_REQUESTED** [review]
### Hello, @maaloum            :wave:   
 Good job so far,  done There are some issues that you still need to work on to go to the next project but you are almost there!
 
 ## Highlights
 * Good job with adding trying to update the controller :+1: 

 ## Required Changes 

- [ ] Check the frontend repo =>  https://github.com/maaloum/hello_rails_fontend/pull/1

 ## Optional suggestions
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
- [ OPTIONAL ] I am writing this as optional because it wasnt asked by the previous reviewer but please in future cases make sure to open the pull request from a feature branch to the develop not master
- N/A



Cheers and Happy coding! clapclapclap
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**